starport/pkg/placeholder/replace.go

```diff
@@ -0,0 +1,125 @@
+package placeholder
+
+import (
+	"context"
+	"strings"
+
+	"github.com/tendermint/starport/starport/pkg/validation"
+)
+
+type iterableStringSet map[string]struct{}
+
+func (set iterableStringSet) Iterate(f func(i int, element string) bool) {
+	i := 0
+	for key := range set {
+		if !f(i, key) {
+			return
+		}
+		i++
+	}
+}
+
+var _ validation.Error = (*ErrMissingPlaceholders)(nil)
+
+type ErrMissingPlaceholders struct {
+	missing iterableStringSet
+}
+
+// Is true if both errors have the same list of missing placeholders.
+func (e *ErrMissingPlaceholders) Is(err error) bool {
+	other, ok := err.(*ErrMissingPlaceholders)
+	if !ok {
+		return false
+	}
+	if len(other.missing) != len(e.missing) {
+		return false
+	}
+	for i := range e.missing {
+		if e.missing[i] != other.missing[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (e *ErrMissingPlaceholders) Error() string {
+	var b strings.Builder
+	b.WriteString("missing placeholders: ")
+	e.missing.Iterate(func(i int, element string) bool {
+		if i > 0 {
+			b.WriteString(", ")
+		}
+		b.WriteString(element)
+		return true
+	})
+	return b.String()
+}
+
+func (e *ErrMissingPlaceholders) ValidationInfo() string {
+	var b strings.Builder
+	b.WriteString("Missing placeholders:\n\n")
+	e.missing.Iterate(func(i int, element string) bool {
+		if i > 0 {
+			b.WriteString("\n")
+		}
+		b.WriteString(element)
+		return true
+	})
+	b.WriteString("\n\n")
+	b.WriteString("Visit https://docs.starport.network/troubleshooting/placeholders.html.")
```

Comment:
to make this pkg more generic, can we make additional info like this url configurable through cmd?

Reply:
do you mean configurable at runtime (e.g. flags or config file)?

Reply:
No not through flags or config. I mean, we can just create a constant with this message inside scaffolder or cmd package and pas it to placeholder.New.

e.g.: placeholder.New(WithDecription(message))